src/components/GotoLineModal.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getActiveSearch, getSelectedSource } from "../selectors";
+import actions from "../actions";
+
+import Modal from "./shared/Modal";
+
+import SearchInput from "./shared/SearchInput";
+
+import type { SourceRecord } from "../reducers/sources";
+import type { SelectSourceOptions } from "../actions/sources";
+
+type GotoLineModalState = {
+  query: ?string
+};
+
+import "./SymbolModal.css";
+
+class GotoLineModal extends Component {
+  state: GotoLineModalState;
+
+  props: {
+    enabled: boolean,
+    selectSource: (string, ?SelectSourceOptions) => void,
+    selectedSource?: SourceRecord,
+    closeActiveSearch: () => void,
+    highlightLineRange: ({ start: number, end: number }) => void,
+    clearHighlightLineRange: () => void
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { query: "" };
+  }
+
+  onClick = (e: SyntheticEvent) => {
+    e.stopPropagation();
+  };
+
+  onChange = (e: SyntheticInputEvent) => {
+    const { selectedSource } = this.props;
+    if (!selectedSource || !selectedSource.get("text")) {
+      return;
+    }
+
+    this.setState({ query: e.target.value });
+  };
+
+  closeModal = () => {
+    this.props.closeActiveSearch();
+    this.props.clearHighlightLineRange();
+  };
+
+  onKeyUp = (e: SyntheticKeyboardEvent) => {
+    e.preventDefault();
+    const { selectSource, selectedSource, enabled } = this.props;
+    const { query } = this.state;
+
+    if (!enabled || !selectedSource) {
+      return;
+    }
+
+    if (e.key === "Enter" && query != null) {
+      const linenumber = parseInt(query.replace(/[^\d+]/g, ""), 10);
+      if (!isNaN(linenumber)) {
+        selectSource(selectedSource.get("id"), { line: linenumber });
+      }
+      this.closeModal();
+    } else if (e.key === "Tab") {
```

Comment:
this is just a style thing, but i tend to prefer early returns

```js
if (e.key === "Enter" && query != null) {
 if (!isNaN(linenumber)) {
   selectSource(selectedSource.get("id"), { line: linenumber })
 }
 return this.closeModal();
}

if (e.key === "Tab") {
  return this.closeModal();
}
```

Reply:
I'll defer to you on that 😃 ! But, as those return values aren't being dispatched on in the caller, I'd prefer 

```js
if (e.key === "Enter" && query != null) {
 if (!isNaN(linenumber)) {
   selectSource(selectedSource.get("id"), { line: linenumber })
 }
 this.closeModal();
 return;
}

if (e.key === "Tab") {
  this.closeModal();
  return;
}
```

When I was moving some logic to this from `SymbolModal.js` I had trouble grokking [this line](https://github.com/devtools-html/debugger.html/blob/master/src/components/SymbolModal.js#L109), because I assumed the `return` value was significant. It can't be, since [updateResults](https://github.com/devtools-html/debugger.html/blob/master/src/components/SymbolModal.js#L131-L145) always returns undefined, but semantically that was unclear.

Is there a technical reason to avoid `else if` in favor of multiple `if` blocks? Like a branching implementation thing?